File: frontend/erp-portfolio-web/src/types/cliente.ts
/**
 * =====================================================================
 * Arquivo....: cliente.ts
 * Versão.....: 1.2.0
 * Data.......: 21/09/2026
 * Descrição..: Tipos do módulo de Clientes, espelhando os DTOs da API
 *              (ClienteRespostaDto, ClienteCriacaoDto, ClienteAtualizacaoDto,
 *              ClienteFiltroDto e ResultadoPaginadoDto).
 * ---------------------------------------------------------------------
 * Histórico de alterações:
 *   1.0.0 - 18/09/2026 - Criação do arquivo.
 *   1.1.0 - 18/09/2026 - Filtros ufs e ativo em ClienteFiltro.
 *   1.2.0 - 21/09/2026 - ResultadoPaginado movido para paginacao.ts (reexportado aqui).
 * =====================================================================
 */

export interface Cliente {
  id: number
  nome: string
  documento: string
  email: string | null
  telefone: string | null
  cidade: string
  uf: string
  ativo: boolean
  /** Data/hora ISO 8601 em UTC. */
  dataCadastro: string
}

export interface ClienteCriacao {
  nome: string
  documento: string
  email: string | null
  telefone: string | null
  cidade: string
  uf: string
}

export interface ClienteAtualizacao extends ClienteCriacao {
  ativo: boolean
}

export interface ClienteFiltro {
  nome?: string
  ufs?: string[]
  /** true = só ativos, false = só inativos, ausente = todos. */
  ativo?: boolean
  pagina: number
  tamanhoPagina: number
}

export type { ResultadoPaginado } from './paginacao'

File: backend/ErpPortfolio.Api/Models/Cliente.cs
// =====================================================================================
// Arquivo....: Cliente.cs
// Versão.....: 1.0.0
// Data.......: 18/09/2026
// Descrição..: Entidade de domínio que representa um cliente (pessoa física ou jurídica).
// -------------------------------------------------------------------------------------
// Banco......: PostgreSQL - erp_portfolio_db
// Tabelas....: public.clientes
// Fontes.....: Mapeada em ErpPortfolioDbContext.Clientes (EF Core / Npgsql).
//              Colunas: id, nome, documento, email, telefone, cidade, uf, ativo,
//              data_cadastro (o mapeamento de nomes fica no DbContext).
// -------------------------------------------------------------------------------------
// Histórico de alterações:
//   1.0.0 - 18/09/2026 - Criação do arquivo.
// =====================================================================================

namespace ErpPortfolio.Api.Models;

public class Cliente
{
    public int Id { get; set; }

    public string Nome { get; set; } = string.Empty;

    /// <summary>CPF (11 dígitos) ou CNPJ (14 caracteres), armazenado sem máscara.</summary>
    public string Documento { get; set; } = string.Empty;

    public string? Email { get; set; }

    public string? Telefone { get; set; }

    public string Cidade { get; set; } = string.Empty;

    /// <summary>Sigla da unidade federativa, sempre em maiúsculas (ex.: SP).</summary>
    public string Uf { get; set; } = string.Empty;

    public bool Ativo { get; set; } = true;

    /// <summary>Data/hora de cadastro em UTC.</summary>
    public DateTime DataCadastro { get; set; }
}
